### PR TITLE
Fixes for changeling transformations. Part 2.

### PR DESCRIPTION
**Pull request by Christmas5**
## About The Pull Request
[A small continuation of #13527.](https://github.com/Skyrat-SS13/Skyrat-tg/pull/13527)

There are things I missed and/or haven't been able to fix in the first PR.
This one adds screams, laughs and age to changeling profiles.

I began working on quirks as well, but I'm not entirely sure how to proceed as copying every single quirk could cause balancing issues. For example, I don't believe a changeling shapeshifting into someone blind should become blind as well.
Someone suggested a toggle button, allowing the changeling to switch between their target's quirks and their own.

Also, the sprite in your examine tab will now shapeshift in sync with your character... and the sprite becomes bald. Not ideal but better than keeping the initial one.

**Issues left:**
- After shapeshifting as a changeling, your sprite in the examine tab will become permanently bald.
- Using lesser form with non-standard body sizes still breaks sprites.
- Clothing overlays do not update properly when using lesser form/human form.
- When using last resort, sometimes the injected egg won't hatch.
- Organs do not update entirely as they should.
- Health scanners can still tell whether you are truly blind and/or deaf, and they also read your true brain traumas.
- The sechud icons for weapon permits do not copy over.

## Changelog

**Changeling profiles save more information about the target.**
- Scream and laugh types.
- Age.
- Quirks.

**A changeling's quirks will now change when shapeshifting.**
Because I'm not yet sure how to balance it, you will not receive all of your targets quirks when shapeshifting. The quirks which you can lose/gain when shapeshifting are defined in _var/list/mimicable_quirks_list_ in _changeling.dm._

`	var/list/mimicable_quirks_list = list(
		"Bad Touch",
		"Sensitive Snout",
		"Ash aspect (Emotes)",
		"Canidae Traits",
		"Excitable!",
		"Feline Traits",
		"Floral aspect (Emotes)",
		"Heterochromatic",
		"Hydra Heads",
		"Oversized",
		"Personal Space",
		"Pseudobulbar Affect",
		"Shifty Eyes",
		"Smooth-Headed",
		"Sparkle aspect (Emotes)",
		"Water aspect (Emotes)",
		"Webbing aspect (Emotes)",
		"Friendly",
	)`

Health analyzers now check if you are a changeling before reading your quirks, and if you are, they will read the quirks of the person you're disguised as, instead of reading your true quirks.

This **only** affects the "Major/Minor Disabilities" rows; scanners will still read your true brain traumas and be able to tell whether you are blind and/or deaf.

:cl:
fix: Changeling transformations now update scream/laugh types, age, as well as certain quirks.
/:cl:

### CODE REVIEW

File: code/modules/antagonists/changeling/changeling.dm [hunk 1]
```diff
@@ -714,6 +717,9 @@
 	user.grad_style = LAZYLISTDUPLICATE(chosen_profile.grad_style)
 	user.grad_color = LAZYLISTDUPLICATE(chosen_profile.grad_color)
 	user.physique = chosen_profile.physique
+	user.selected_scream = new chosen_profile.scream_type
```
Comment (Zonespace27) [inline code/modules/antagonists/changeling/changeling.dm:749]: are the lingering scream/laugh references deleted anywhere else?

Comment (Christmas5) [reply code/modules/antagonists/changeling/changeling.dm:749]: Uh-oh, no I don't believe so. Do you mean that when a new scream/laugh is made for the character, the old one doesn't get removed?

Would it make sense to select and delete the current one before making a new one? If so, how would I go about that?

**Comment by Kush1Push1** [comment]
Thanks for working with it

File: code/modules/antagonists/changeling/changeling.dm (hunk 1 above)
Comment (Christmas5) [reply code/modules/antagonists/changeling/changeling.dm:749]: I just used qdel(), let me know if that's not enough.

**Comment by Zergspower** [comment]
Hilariously enough - and this really isnt for you to deal with - but OOC always seems to match whomever - so FT would transfer but uh lol
image

**Comment by Zonespace27** [comment]
> Hilariously enough - and this really isnt for you to deal with - but OOC always seems to match whomever - so FT would transfer but uh lol image

it checks your client, that won't be changing

**Comment by Zergspower** [comment]
Nah, just found it amusing enough to share

**Comment by Christmas5** [comment]
@Zonespace27 I don't believe any of the failing checks are related to my changes.

/ Windows Build gives a warning
`code\modules\mob\living\carbon\human\human_update_icons.dm:162:warning: handled_by_bodytype: variable defined but not used`

/ Integration Tests all give
`Error: Assertion failed: The Mauler was shot with a laser, but didn't take any damage.`
`Error: Assertion failed: The Ultra AC 2 was shot with a laser, but didn't take any damage.`
`Error: Assertion failed: The Mauler was shot with a bullet, but didn't take any damage.`
`Error: Assertion failed: The Ultra AC 2 was shot with a bullet, but didn't take any damage.`
`Error: tournament gangster's r_pocket is invalid! Could not equip a /obj/item/ammo_box/c10mm into that slot.`
`Error: [17:44:22] Runtime in outfit_sanity.dm,51: Cannot execute null.on outfit equip().`

/ Integration Tests (kilostation) in particular also gives
`Error: /mob/living/simple_animal/hostile/retaliate/snake hard deleted 1 times out of a total del count of 4`

The PR appears to work fine on my local copy. Does it have to pass checks before getting merged?

**Comment by Christmas5** [comment]
> Hilariously enough - and this really isnt for you to deal with - but OOC always seems to match whomever - so FT would transfer but uh lol image

Lmao I completely missed that, thank you for catching it!

**Comment by Zonespace27** [comment]
> @Zonespace27 I don't believe any of the failing checks are related to my changes.
> 
> / Windows Build gives a warning `code\modules\mob\living\carbon\human\human_update_icons.dm:162:warning: handled_by_bodytype: variable defined but not used`
> 
> / Integration Tests all give `Error: Assertion failed: The Mauler was shot with a laser, but didn't take any damage.` `Error: Assertion failed: The Ultra AC 2 was shot with a laser, but didn't take any damage.` `Error: Assertion failed: The Mauler was shot with a bullet, but didn't take any damage.` `Error: Assertion failed: The Ultra AC 2 was shot with a bullet, but didn't take any damage.` `Error: tournament gangster's r_pocket is invalid! Could not equip a /obj/item/ammo_box/c10mm into that slot.` `Error: [17:44:22] Runtime in outfit_sanity.dm,51: Cannot execute null.on outfit equip().`
> 
> / Integration Tests (kilostation) in particular also gives `Error: /mob/living/simple_animal/hostile/retaliate/snake hard deleted 1 times out of a total del count of 4`
> 
> The PR appears to work fine on my local copy. Does it have to pass checks before getting merged?

these are all repo issues being ironned out

**Comment by Christmas5** [comment]
> > @Zonespace27 I don't believe any of the failing checks are related to my changes.
> > / Windows Build gives a warning `code\modules\mob\living\carbon\human\human_update_icons.dm:162:warning: handled_by_bodytype: variable defined but not used`
> > / Integration Tests all give `Error: Assertion failed: The Mauler was shot with a laser, but didn't take any damage.` `Error: Assertion failed: The Ultra AC 2 was shot with a laser, but didn't take any damage.` `Error: Assertion failed: The Mauler was shot with a bullet, but didn't take any damage.` `Error: Assertion failed: The Ultra AC 2 was shot with a bullet, but didn't take any damage.` `Error: tournament gangster's r_pocket is invalid! Could not equip a /obj/item/ammo_box/c10mm into that slot.` `Error: [17:44:22] Runtime in outfit_sanity.dm,51: Cannot execute null.on outfit equip().`
> > / Integration Tests (kilostation) in particular also gives `Error: /mob/living/simple_animal/hostile/retaliate/snake hard deleted 1 times out of a total del count of 4`
> > The PR appears to work fine on my local copy. Does it have to pass checks before getting merged?
> 
> these are all repo issues being ironned out

Then it should be ready for merging.

**Pull request merged**